Reference sensitive data from secrets and config maps (#1024)
wandb/docs: content/guides/hosting/hosting-options/self-managed/kubernetes-operator/_index.md

    secretKey: HDKYe4Q...JAp1YyjysnX
```

> The `kmsKey` must be `null`.

Reviewer: I'm confused -- if it must be null, then why have the key at all? In the example for AWS above, the `kmsKey` is the empty string. Must it be empty for S3 and null for S3-compatible storage that is not on AWS?

> The `kmsKey` must be `null`.

Reviewer: ```suggestion
For S3-compatible storage hosted outside of AWS, `kmsKey` must be `null`.
```

To reference `accessKey` and `secretKey` from a secret:
```yaml
global:
  bucket:
    # Example values, replace with your own
    provider: s3
    name: storage.example.com
    kmsKey: null
    path: wandb
    region: default
    secret:

>       secretName: wandb-bucket

Reviewer 2: suggestion: `bucket-secret` 

Just to ensure it will not clash with any secret we create by default. All of our stuff is prefixed with `wandb`

      accessKeyName: ACCESS_KEY
      secretKeyName: SECRET_KEY
```

### MySQL

```yaml
global:
   mysql:
     # Example values, replace with your own
     host: db.example.com
     port: 3306
     database: wandb_local
     user: wandb
     password: 8wtX6cJH...ZcUarK4zZGjpV 
```

To reference the `password` from a secret:
```yaml
global:
   mysql:
     # Example values, replace with your own
     host: db.example.com
     port: 3306
     database: wandb_local
     user: wandb
     passwordSecret:

>        name: mysql-wandb-credentials

Reviewer 2: Same here but just to standardize the secret names, I suggest `database-secret`
Using `mysql-secret` might clash with some MySQL from some other MySQL deployment.

       passwordKey: MYSQL_WANDB_PASSWORD
```

### License

```yaml
global:
  # Example license, replace with your own
  license: eyJhbGnUzaHgyQjQy...VFnPS_KETXg1hi
```

To reference the `license` from a secret:
```yaml
global:
  licenseSecret:

>     name: customer-wandb-license

Reviewer 2: Same... `license-secret`

    oidc:
      clientId: ""
      secret: ""

>       # Only include if your IdP requires it.

Reviewer: Do we know of any major IdPs that do or don't require it? Okta? AAD? GCP?

      authMethod: ""
      issuer: ""
```

> `authMethod` is optional. 

Reviewer: The comment previously seems to imply that it is not optional, either your IdP requires it (it's not optional) or it doesn't (then does it even evaluate the field?)

    -----END CERTIFICATE-----
```

CA certificates can also be stored in a ConfigMap:
```yaml
global:
  caCertsConfigMap: custom-ca-certs
```

The ConfigMap must look like this:
```yaml
apiVersion: v1
kind: ConfigMap
metadata:
  name: custom-ca-certs
data:

>   ca-cert1.crt: |

Reviewer: Do we have any best practices around compound certs? For example, they can become unwieldy and poor performing in a clustered environment with many nodes, if they share a compound certificate.

    -----BEGIN CERTIFICATE-----
    ...
    -----END CERTIFICATE-----
  ca-cert2.crt: |
    -----BEGIN CERTIFICATE-----
    ...
    -----END CERTIFICATE-----
```

> **Important:**

Reviewer: Also, this is no longer the righrt way to do an admonition . Instead, use the Hugo shortcode:
```
{{% alert %}}
...
{{% /alert %}}
```

> If using a ConfigMap, each key in the ConfigMap must end with `.crt` (e.g., `my-cert.crt` or `ca-cert1.crt`). This naming convention is required for `update-ca-certificates` to parse and add each certificate to the system CA store.

Reviewer: Please fix this and following instances of the same feedback. I took two years of Latin in high school, but most of our readers probably did not.

> **Important:**
> If using a ConfigMap, each key in the ConfigMap must end with `.crt` (e.g., `my-cert.crt` or `ca-cert1.crt`). This naming convention is required for `update-ca-certificates` to parse and add each certificate to the system CA store.

Reviewer: ```suggestion
{{% alert %}}
If using a ConfigMap, each key in the ConfigMap must end with `.crt` (for example, `my-cert.crt` or `ca-cert1.crt`). This naming convention is required for `update-ca-certificates` to parse and add each certificate to the system CA store.
{{% /alert %}}
```

{{% /alert %}}

> Example: To configure the application pod, add a section `app` to your configuration:

Reviewer: ```suggestion
For example, to configure the application pod, add a section `app` to your configuration:
```

      readOnlyRootFilesystem: false
      allowPrivilegeEscalation: false 
```

> The same concept applies to `console`, `weave`, `weave-trace` and `parquet`.

Reviewer: ```suggestion

The same concept applies to `console`, `weave`, `weave-trace` and `parquet`.
```

> The same concept applies to `console`, `weave`, `weave-trace`, `flat-run-fields-updater` and `parquet`.

Reviewer 2: Do we need to mention the `flat-run-fields-updater`?

  -----END CERTIFICATE-----
```

CA certificates can also be stored in a ConfigMap:
```yaml
caCertsConfigMap: custom-ca-certs
```

The ConfigMap must look like this:
```yaml
apiVersion: v1
kind: ConfigMap
metadata:
  name: custom-ca-certs
data:
  ca-cert1.crt: |
    -----BEGIN CERTIFICATE-----
    ...
    -----END CERTIFICATE-----
  ca-cert2.crt: |
    -----BEGIN CERTIFICATE-----
    ...
    -----END CERTIFICATE-----
```

> **Important:**

Reviewer: As above, please also update this admonition.

> **Important:**

Reviewer: ```suggestion
```

> Each key in the ConfigMap must end with `.crt` (e.g., `my-cert.crt` or `ca-cert1.crt`). This naming convention is required for `update-ca-certificates` to parse and add each certificate to the system CA store.

Reviewer: ```suggestion
{{% alert %}}
Each key in the ConfigMap must end with `.crt` (e.g., `my-cert.crt` or `ca-cert1.crt`). This naming convention is required for `update-ca-certificates` to parse and add each certificate to the system CA store.
{{% /alert %}}
```
